doc/source/development/rfc/rfc103_schema_open_option.rst

```diff
@@ -0,0 +1,275 @@
+            "type": "array",
+            "items": {
+                "$ref": "#/definitions/field"
+            }
+            }
+        },
+        "required": [
+            "name",
+            "fields"
+        ],
+        "additionalProperties": false
+        },
+        "field": {
+        "description": "The field definition",
+        "additionalProperties": true,
+        "type": "object",
+        "properties": {
+            "name": {
+            "type": "string"
+            }
+        },
+        "anyOf": [
+            {
+            "type": "object",
+            "properties": {
+                "type": {
+                "$ref": "#/definitions/fieldType"
+                },
+                "subType": {
+                "$ref": "#/definitions/fieldSubType"
+                },
+                "width": {
+                "type": "integer"
+                },
+                "precision": {
+                "type": "integer"
+                }
+            },
+            "required": [
+                "type"
+            ]
+            },
+            {
+            "description": "The new name of the field",
+            "newName": {
+                "type": "string"
+            },
+            "required": [
+                "newName"
+            ]
+            }
+        ],
+        "required": [
+            "name"
+        ]
+        },
+        "fieldType": {
+        "enum": [
+            "Integer",
+            "Integer64",
+            "Real",
+            "String",
+            "Binary",
+            "IntegerList",
+            "Integer64List",
+            "RealList",
+            "StringList",
+            "Date",
+            "Time",
+            "DateTime"
+        ]
+        },
+        "fieldSubType": {
+        "enum": [
+            "None",
+            "Boolean",
+            "Int16",
+            "Float32",
+            "JSON",
+            "UUID"
+        ]
+        }
+    }
+    }
+
+Here is an example of a schema document that will be used to override the fields type and the name of a dataset using the default "Patch" mode:
+
+.. code-block:: json
+
+   {
+    "layers": [
+        {
+        "name": "layer1",
+        "fields": [
+            {
+            "name": "field1",
+            "type": "String",
+            "subType": "JSON"
+            },
+            {
+            "name": "field2",
+            "newName": "new_field2"
+            }
+        ]
+        }
+    ]
+    }
+
+
+In case of multi-layered datasets, the schema will be specified as a list of layers, each with its own fields definition and Patch/Full mode:
+
+.. code-block:: json
+
+    {
+    "layers":[
+        {
+        "name": "layer1",
+        "schema_type": "Full",
+        "fields": [
+            {
+            "name": "field1",
+            "type": "String",
+            "subType": "JSON"
+            },
+            {
+            "name": "field2",
+            "newName": "new_field2"
```

Comment:
That makes sense to me. As an alternative we could treat the fields as an object such as:
```json
"fields" : { "field1" { "name" : "field1_renamed" ...}}
```
that would mean that we abandon the idea to use the output of `ogrinfo -json` as a template for the schema, but perhaps we have already lost that train.

Reply:
I hope not to lose everything from the `ogrinfo -json`. It would be rather demanding to write a working schema from a scratch, and having a separate option or utility for printing the default schema feels like duplication. But rather duplication than leaving users alone.

Reply:
@elpaso yeah, the original sin (so to speak) is that the OGR schema is represented as an array of fields (this is OGR's internal model) instead of as a dictionary. Thus we have to reference fields by "the item in the `fields` array where `name` is `val`" instead of just "`fields[val]`". I think it makes more sense to accept that than to change it.